### PR TITLE
feat(crashlytics): add new APIs `checkForUnsentReports`, `deleteUnsentReports`,`didCrashOnPreviousExecution`,`sendUnsentReports`

### PR DESCRIPTION
**Pull request by russellwheatley**
### Description

Additional Crashlytics API, including:

- checkForUnsentReports()
- deleteUnsentReports()
- didCrashOnPreviousExecution()
- sendUnsentReports()

### Notes
- I've created Android error messages for `checkForUnsentReports` which I'm not sure about especially as there is no equivalent mechanism on the iOS side. I welcome thoughts on what to do about that.

### Related issues

None that I'm aware of.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [X] Yes
- My change supports the following platforms;
  - [X] `Android`
  - [X] `iOS`
- My change includes tests;
  - [X] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [X] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [X] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Comment by mikehardy** [comment]
Those typedoc files are huge, are they necessary?

File: tests/ios/Podfile.lock [hunk 1]
```diff
@@ -848,14 +848,14 @@ SPEC CHECKSUMS:
...
-  RNFBRemoteConfig: fb46148a3b985697f3538b8c4d3e586a14bb4570
+  RNFBPerf: 3d185f43c89d4159649c6a35813c337857a2e941
+  RNFBRemoteConfig: 96917b7defbb474ffc774f43454686de7df09a9b
   RNFBStorage: ee1bafd9e83d30feaa4af266f2782f109284be9a
   Yoga: 3ebccbdd559724312790e7742142d062476b698e
 
 PODFILE CHECKSUM: 250927a7e70dee93adbb105e4000258dc7be161b
 
-COCOAPODS: 1.9.3
+COCOAPODS: 1.9.1
```
Comment (mikehardy) [inline tests/ios/Podfile.lock:861]: You should update your local cocoapods ;-)

File: tests/ios/Podfile.lock [hunk 2]
```diff
@@ -569,11 +569,11 @@ PODS:
     - Firebase/MLVisionTextModel (= 6.28.1)
     - React
     - RNFBApp
-  - RNFBPerf (7.2.2):
+  - RNFBPerf (7.2.3):
     - Firebase/Performance (= 6.28.1)
     - React
     - RNFBApp
-  - RNFBRemoteConfig (7.2.2):
+  - RNFBRemoteConfig (8.0.0):
```
Comment (mikehardy) [inline tests/ios/Podfile.lock:576]: Does this imply that the whole test project should be updated via Podfile / package.json as well?

File: packages/crashlytics/lib/index.d.ts [hunk 3]
```diff
@@ -78,6 +78,51 @@ export namespace FirebaseCrashlyticsTypes {
...
      */
     isCrashlyticsCollectionEnabled: true;
+    /**
+     * Determines whether there are any unsent crash reports cached on the device. The callback only executes
+     * if automatic data collection is disabled.
+     *
+     * #### Example
+     *
+     * ```js
+     * async checkReports(){
```
Comment (mikehardy) [inline packages/crashlytics/lib/index.d.ts:88]:
```suggestion
     * async checkReports() {
```

File: packages/crashlytics/lib/index.d.ts [hunk 4]
```diff
@@ -78,6 +78,51 @@ export namespace FirebaseCrashlyticsTypes {
...
+     *
+     */
+    deleteUnsentReports(): void;
+    /**
+     * Returns a boolean value indicating whether the app crashed during the previous execution.
+     *
+     * #### Example
+     *
+     * ```js
+     * async didCrashPreviously(){
```
Comment (mikehardy) [inline packages/crashlytics/lib/index.d.ts:116]:
```suggestion
     * async didCrashPreviously() {
```

File: packages/crashlytics/android/src/main/java/io/invertase/firebase/crashlytics/ReactNativeFirebaseCrashlyticsModule.java [hunk 5]
```diff
@@ -35,6 +35,23 @@
     super(reactContext, TAG);
   }
 
+  @ReactMethod
+  public void checkForUnsentReports(Promise promise){
+    FirebaseCrashlytics.getInstance().checkForUnsentReports().addOnCompleteListener(task -> {
+      if(task.isSuccessful()){
```
Comment (mikehardy) [inline packages/crashlytics/android/src/main/java/io/invertase/firebase/crashlytics/ReactNativeFirebaseCrashlyticsModule.java:41]:
```suggestion
      if(task.isSuccessful()) {
```

File: tests/ios/Podfile.lock (hunk 1 above)
Comment (russellwheatley) [reply tests/ios/Podfile.lock]: done :)

File: tests/ios/Podfile.lock (hunk 2 above)
Comment (russellwheatley) [reply tests/ios/Podfile.lock:576]: tis a good question. My impression was that running `pod install` in ios directory was the best way to keep the lockfile up to date, but I'll defer that wisdom to @Salakar

**Comment by russellwheatley** [comment]
@mikehardy Apologies if I misunderstood you, but the typedocs are necessary for generating docs for the documentation website. They're huge but I'm not expecting anyone to review them :)

File: tests/ios/Podfile.lock (hunk 2 above)
Comment (Salakar) [reply tests/ios/Podfile.lock:576]: RNFB pod versions are importing the `package.json` file and reading the version from there, no manual updates necessary. `pod install` in the tests/ios directory is all that is needed

File: packages/crashlytics/lib/index.d.ts [hunk 6]
```diff
@@ -78,6 +78,51 @@ export namespace FirebaseCrashlyticsTypes {
...
+     * disabled.
+     *
+     * #### Example
+     *
+     * ```js
+     * firebase.crashlytics().deleteUnsentReports();
+     * ```
+     *
+     */
+    deleteUnsentReports(): void;
```
Comment (Salakar) [inline packages/crashlytics/lib/index.d.ts:109]:
```suggestion
    deleteUnsentReports(): Promise<void>;
```

As the native call is async lets make this one return a Promise (JS method also needs updating)

**Pull request merged**